tests/unittest_brain.py

```diff
@@ -2020,5 +2020,24 @@ class Other:
     assert isinstance(name[0], astroid.Unknown)
 
 
+def test_no_recursionerror_on_self_referential_length_check():
+    """
+    Regression test for https://github.com/PyCQA/astroid/issues/777
+    """
+    with test_utils.pytest_assert_not_raises(RecursionError):
```

Comment:
Let's remove this check entirely and simply rely on checking for inference errors. Checking against `RecursionError` itself does not provide any value.
```
with pytest.raises(InferenceError):
     next(node.infer())
```